### PR TITLE
feat: refactor benchmark documentation

### PR DESCRIPTION
**Pull request by antomys**
# This merge request updates and improves benchmark documentation. 

Changes include:
- Restructured content for better readability;
- Updated outdated information;

These changes aim to make benchmark documentation more  user-friendly.

@andanteyk, your review would be greatly appreciated. Please pay particular attention to:

- Accuracy of the updated information
- Clarity of the new explanations
- Completeness of the documentation

Let me know if you need any clarification or have suggestions for further improvements.

### CODE REVIEW

**Comment by andanteyk** [comment]
Thanks for your work.

For future reference: did you create this table manually?

If there is a command to create it in bulk, I thought it would be useful to include that in the readme.

File: docs/BenchmarkResult.md [hunk 1]
```diff
@@ -1,170 +1,526 @@
...
+| Chunk                    | SystemChunk                    |       704.1107 ns |      2.3762 ns |      1.9843 ns | 0.5283 |      - |    6632 B |
+| Chunk                    | HandcraftedChunk               |       853.0973 ns |      7.8673 ns |      7.3591 ns | 0.5922 | 0.0019 |    7440 B |
+| Chunk                    | SpanChunk                      |     1,136.6704 ns |     12.9198 ns |     12.0852 ns | 0.5283 |      - |    6632 B |
+
+### Concat
+| Type                          | Method                         |              Mean |          Error |         StdDev |   Gen0 |   Gen1 | Allocated |
+|-------------------------------|--------------------------------|------------------:|---------------:|---------------:|-------:|-------:|----------:|
+| Concat                   | SystemConcat                   |     3,456.7250 ns |     18.5042 ns |     17.3088 ns | 0.0076 |      - |     120 B |
+| Concat                   | SpanConcat                     |     6,078.0059 ns |     18.3870 ns |     17.1992 ns |      - |      - |         - |
+| Contains                 | SpanContains                   |        47.2104 ns |      0.2871 ns |      0.2685 ns |      - |      - |         - |
```
Comment (andanteyk) [inline docs/BenchmarkResult.md:166]: `Contains` is not necessary here.

**Comment by antomys** [comment]
Thank you for the feedback. I appreciate your interest.

You're right, I created this table manually. However, your suggestion to include a command for bulk creation in the README is excellent. I'll look into automated methods to generate the table and update the documentation accordingly in upcoming pull requests. This should make the process more efficient for future contributors.

File: docs/BenchmarkResult.md (hunk 1 above)
Comment (antomys) [reply docs/BenchmarkResult.md:166]: Fixed, thanks

**Review by andanteyk — APPROVED** [review]

**Pull request merged**